src/backend-proxy.js

```diff
@@ -69,6 +69,14 @@ function createHandler({request, response, next, options, backendHttpOptions}) {
 		const contentType = (backendResponse.headers['content-type'] || '').toLowerCase();
 
 		if (contentType === options.requiredContentType || contentType === `${options.requiredContentType}; charset=utf-8`) {
+			// Supplement response with headers from Backend, if needed
+			if (Array.isArray(options.backendHeaders) && backendResponse.headers) {
+				for (const header in backendResponse.headers) {
+					if (options.backendHeaders.includes(header)) {
+						response.set(header, backendResponse.headers[header]);
+					}
+				}
```

Comment:
If you invert the logic here then you can switch from `O(n * m)` to `O(n)`.

The `backendResponse.headers` is an object so you can do lookups.
Iterate the `options.backendHeader` and copy those that exist on `backendResponse.headers`.

```suggestion
                                options.backendHeaders.filter(header => backendResponse.headers[header])
                                       .forEach(header => response.set(header, backendResponse.headers[header]);
```

_(You might need to fix the linting on this as i've eyeballed the whitespace in github)_